source/AI.cpp

```diff
@@ -3285,21 +3285,22 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
-			
-			double closest = numeric_limits<double>::infinity();
+			double best = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
 			bool foundAnything = false;
+			bool distancePriority = Preferences::Has("Boarding target priority");
 			for(const shared_ptr<Ship> &other : ships)
 				if(CanBoard(ship, *other))
 				{
 					if(shift && !other->IsYours())
 						continue;
 					
 					bool isEnemy = other->GetGovernment()->IsEnemy(ship.GetGovernment());
-					double d = other->Position().Distance(ship.Position());
-					if((isEnemy && !foundEnemy) || (d < closest && isEnemy == foundEnemy))
+					double b = distancePriority ? other->Position().Distance(ship.Position()) : -other->Cost();
+
+					if((isEnemy && !foundEnemy) || (b < best && isEnemy == foundEnemy))
```

Comment:
Is this PR also tested with boarding for disabled escorts and with boarding for repairing disabled friendly ships? (Which I believe are both handled by the same code.)

Reply:
it works exactly the same way indeed, so much so that I don't think it's needed (holding shift means all ships are looked at and ennemy ones are just ignored) meaning it'll behave in the same way but I can test it it's just harder to get my fleet to get rekt, guess it's time to buy variety + remove guns xD